syntax = "proto3";
    
service Streaming {
    rpc GetRange(GetRangeRequest) returns (stream GetRangeResponse) {}
rpc SumNumbers(stream SumNumbersRequest) returns (SumNumbersResponse) {}
rpc DoubleNumbers(stream DoubleNumber) returns (stream DoubleNumber) {}
}

message SumNumbersResponse {
    int64 total = 1;
}

message DoubleNumber {
    int64 number = 1;
}

message GetRangeResponse {
    int64 value = 1;
}

message GetRangeRequest {
    int64 min_value = 1;
int64 max_value = 2;
}

message SumNumbersRequest {
    int64 number = 1;
}